arch/cpu/nrf/sys/soc-rtc.c

```diff
@@ -0,0 +1,212 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+/*---------------------------------------------------------------------------*/
+/**
+ * \addtogroup nrf-sys
+ * @{
+ *
+ * \file
+ * Implementation of the NRF RTC driver
+ */
+/*---------------------------------------------------------------------------*/
+#include "contiki.h"
+#include "clock.h"
+
+#include "soc-rtc.h"
+
+#include "etimer.h"
+#include "lpm.h"
+#include "rtimer.h"
+
+#include "nrfx_clock.h"
+#include "nrfx_rtc.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "sys/log.h"
+#define LOG_MODULE "RTC"
+#define LOG_LEVEL LOG_LEVEL_INFO
+
+/*---------------------------------------------------------------------------*/
+/* Prototype of a function in clock.c. Called every time the handler fires */
+void clock_update(void);
+
+/*---------------------------------------------------------------------------*/
+#define COMPARE_INCREMENT (RTIMER_SECOND / CLOCK_SECOND)
+#define MULTIPLE_256_MASK 0xFFFFFF00
+/*---------------------------------------------------------------------------*/
+#if CLOCK_SIZE != 4
+#error CLOCK_CONF_SIZE must be 4 (32 bit)
+#endif
+#if RTIMER_SECOND != 32768
+#error RIMTER_SECOND must be 32768 when using the NRF_RTC
+#endif
+
+#ifdef NRF_CLOCK_CONF_RTC_INSTANCE
+#define NRF_CLOCK_RTC_INSTANCE NRF_CLOCK_CONF_RTC_INSTANCE
+#else
+#define NRF_CLOCK_RTC_INSTANCE 0
+#endif
+
+/* Prototype of a function in clock.c. Called every time the handler fires */
+void clock_update(void);
+/*---------------------------------------------------------------------------*/
+/**< RTC instance used for platform clock */
+static const nrfx_rtc_t rtc = NRFX_RTC_INSTANCE(NRF_CLOCK_RTC_INSTANCE);
+/*---------------------------------------------------------------------------*/
+static rtimer_clock_t last_isr_time = 0;
```

Comment:
Aren't static variables automatically initialized to 0 in C, so these RHSes can be omitted?

Reply:
this is correct, not necessary.